File: leetcode-submissions/3254_Find_the_Power_of_K-Size_Subarrays_I.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        vector<int> ans(nums.size()-k+1, -1);
        for (int i = 0; i <= nums.size() - k; i++) {
            int locmax = INT_MIN;
            bool hasMax = true;
            for (int j = i; j < i + k; j++) {
                // Check for increasing sequence condition
                if (j < i + k - 1 && nums[j] >= nums[j + 1]) {
                    hasMax = false;
                    break;
                }
                locmax = max(locmax, nums[j]);
            }
            if (hasMax) {
                ans[i] = locmax;
            }
        }
        return ans;
    }
};

File: leetcode-submissions/3254_Find_the_Power_of_K-Size_Subarrays_I.ts
function resultsArray(nums: number[], k: number): number[] {
    let n:number =nums.length;
    const ans:number[]= Array.from({length:n-k+1},()=>-1);
    let count:number =1;
    for(let i:number =1;i<k;i++){
        if(nums[i]===nums[i-1]){
            count++;
        }else{
            count=1;
        }
    }
    let i:number=1,j:number=k;
    while(j<n){
        if(nums[j]===nums[j-1]+1){
            count++;
        }else{
            count =1;
        }
        if(count>=k){
            ans[i]=nums[j];
        }
        i++;
        j++;
    }
return ans;
};